pkg/proxy/util/util.go

```diff
@@ -70,6 +69,12 @@ func CreateYamlFile(path string, fileName string, Logger *zap.Logger) (bool, err
 		return false, err
 	}
 	if _, err := os.Stat(yamlPath); err != nil {
+		// Temporarily set umask to allow broader permissions
+		err = utils.SetUmask(0)
```

Comment:
why are you setting it as 0 here? It just sets it to the original permissions of the program right?

Reply:
That would be incorrect. Setting Umask to 0 temporarily  before creating the file, allows the file to have the  maximum permissions (read, write, and execute) for the owner, group, and others, thus effectively removing the need for explicitly using the chmod command to handle the file permissions